### PR TITLE
fix(smolvla): correct loss normalization for padded actions

### LINKED ISSUE #3433: Incorrect loss normalization when masking padded actions in SmolVLA (Same issue as ACT)
### Ticket Type

🐛 Bug Report (Something isn't working)

### Environment & System Info

```Shell
- LeRobot version: 0.5.2
- Platform: Linux-5.14.0-570.52.1.el9_6.x86_64-x86_64-with-glibc2.34
- Python version: 3.13.11
- Huggingface Hub version: 1.10.1
- Transformers version: N/A
- Datasets version: 4.8.4
- Numpy version: 2.2.6
- FFmpeg version: 6.0.1
- PyTorch version: 2.10.0+cu128
- Is PyTorch built with CUDA support?: True
- Cuda version: 12.8
- GPU model: NVIDIA H100 NVL
- Using GPU in script?: <fill in>
- lerobot scripts: ['lerobot-calibrate', 'lerobot-dataset-viz', 'lerobot-edit-dataset', 'lerobot-eval', 'lerobot-find-cameras', 'lerobot-find-joint-limits', 'lerobot-find-port', 'lerobot-imgtransform-viz', 'lerobot-info', 'lerobot-record', 'lerobot-replay', 'lerobot-setup-can', 'lerobot-setup-motors', 'lerobot-teleoperate', 'lerobot-train', 'lerobot-train-tokenizer']
```

### Description

Same bug as #3353 (originally filed against ACT), but in SmolVLA. #3377 fixed the normalization for ACT, Diffusion, and MultiTaskDiT, but SmolVLA was intentionally left out of that PR's scope.

`modeling_smolvla.py` applies the `action_is_pad` mask to zero out padded timesteps, then reduces with `.mean()`. Because `.mean()` divides by the total number of elements (including the zeroed-out padding), the per-scalar denominator is larger than the number of valid entries, so the reported loss (and therefore the effective gradient signal) is underestimated proportional to the padding fraction. The larger the padding, the larger the dilution.

### Context & Reproduction

1. Train SmolVLA for a handful of steps on any dataset that produces padded action chunks:
   ```bash
   lerobot-train \
       --policy.type=smolvla \
       --policy.vlm_model_name=HuggingFaceTB/SmolVLM2-500M-Video-Instruct \
       --policy.chunk_size=40 --policy.n_action_steps=40 --policy.device=cuda \
       --dataset.repo_id=lerobot/aloha_sim_transfer_cube_human \
       --dataset.episodes="[0,1]" --batch_size=2 --steps=30 \
       --log_freq=1 --save_checkpoint=false --eval_freq=0 --wandb.enable=false
   ```
2. Capture the per-step `loss:` values.
3. Apply the patch above to `modeling_smolvla.py`, re-run with the same seed/config, and compare — `loss(HEAD) / loss(patched) < 1` for any batch with padding present.

### Relevant logs or stack trace

```Shell

```

### Checklist

- [x] I have searched existing tickets to ensure this isn't a duplicate.
- [x] I am using the latest version of the `main` branch.
- [x] I have verified this is not an environment-specific problem.

### Additional Info / Workarounds

- Related: issue #3353 (original ACT report by @harshab1701), PR #3377 (fixed ACT / Diffusion / MultiTaskDiT). SmolVLA was deliberately kept out of #3377's scope.
- Workaround for users running SmolVLA training today: monkey-patch the `compute_loss` method (or override the policy class) to apply the per-scalar-mean formula above.

### PR DESCRIPTION
**Pull request by whats2000**
## Title

fix(smolvla): correct loss normalization for padded actions

## Summary / Motivation

Same dilution bug as #3353 (originally filed for ACT), applied to SmolVLA. PR #3377 fixed ACT, Diffusion, and MultiTaskDiT but left SmolVLA out of scope — this PR extends the same per-scalar-mean pattern to `modeling_smolvla.py`.

Pre-patch, `losses.mean()` (and `losses.mean(dim=(1, 2))`) divides by the total number of elements — including padded timesteps that `action_is_pad` has already zeroed out. The denominator is larger than the number of valid entries, so the loss is silently rescaled by `1 − padding_fraction`. Fix: divide by `(~actions_is_pad).sum() * losses.shape[-1]` instead.

## Related issues

- Fixes / Closes: #3433
- Related: #3353 (original ACT report), #3377 (fix for ACT / Diffusion / MultiTaskDiT)

## What changed

- `src/lerobot/policies/smolvla/modeling_smolvla.py`: both reduction paths (`reduction == "none"` and scalar default) normalize over valid scalar entries when `actions_is_pad` is set. `clamp_min(1)` preserves the all-padded-batch edge case (0/1 = 0).
- `actions_is_pad is None` -> unchanged behavior (`losses.mean()` fallback).
- No API change. Loss *magnitude* changes when padding is present, so tuned LR / early-stop recipes may need re-tuning -> noting for downstream users, same caveat as #3377.

## How was this tested (or how to run locally)

- `uv run ruff format` and `uv run ruff check` pass clean on the changed file.
- Empirical A/B comparison on current HEAD (variant A = pre-patch, variant B = this fix), 30 training steps, fixed seed, `lerobot/aloha_sim_transfer_cube_human`, `chunk_size=40`, `n_action_steps=40`, `batch_size=2`, single NVIDIA GB200:

  ```
  mean loss_A / loss_B = 0.9672    stdev=0.0884
  ```

  Direction and magnitude consistent with #3377's `loss_A / loss_C ≈ 0.9603` for ACT on the same dataset. Heavier-padding setups (60–70% padded horizon, per #3353) will show a larger gap.

- Repro command:

  ```bash
  lerobot-train \
    --policy.type=smolvla \
    --policy.vlm_model_name=HuggingFaceTB/SmolVLM2-500M-Video-Instruct \
    --policy.chunk_size=40 --policy.n_action_steps=40 --policy.device=cuda \
    --dataset.repo_id=lerobot/aloha_sim_transfer_cube_human \
    --dataset.episodes="[0,1]" --batch_size=2 --steps=30 \
    --log_freq=1 --save_checkpoint=false --eval_freq=0 --wandb.enable=false
  ```

## Checklist (required before merge)

- [x] Linting/formatting run (`pre-commit run -a`) hooks don't apply to this file.
- [x] All tests pass locally (`pytest`) 
- [x] Documentation updated: N/A
- [ ] CI is green: Will confirm after PR opens.
- [ ] Community Review: TBR

## Reviewer notes

- Core change is ~10 lines in one function; logic mirrors what #3377 applied to ACT / Diffusion / MultiTaskDiT, including the `clamp_min(1)` safety and the `None`-branch fallback.
- Worth a sanity check on the `reduction == "none"` path — per-sample loss becomes `losses.sum(dim=(1, 2)) / num_valid` with `num_valid` broadcasting per-batch-row.

### CODE REVIEW

**Review by imstevenpmwork — APPROVED** [review]
LGTM, thanks!
